mlflow/tracking/fluent.py

```diff
@@ -388,11 +444,19 @@ def start_run(
             ).format(active_run_stack[0].info.run_id)
         )
     client = MlflowClient()
+
+    # Get SGC job run ID tag key for run resumption if applicable
+    sgc_job_run_id_tag_key = _get_sgc_job_run_id_tag_key()
+
     if run_id:
         existing_run_id = run_id
     elif run_id := MLFLOW_RUN_ID.get():
         existing_run_id = run_id
         del os.environ[MLFLOW_RUN_ID.name]
+    elif sgc_job_run_id_tag_key:
```

Comment:
```suggestion
    elif sgc_job_run_id_tag_key := _get_sgc_job_run_id_tag_key():
```

Let's move the call here to avoid a useless call.